wfe/web-front-end.go

```diff
@@ -136,35 +159,23 @@ func verifyPOST(request *http.Request) ([]byte, jose.JsonWebKey, error) {
 	// *anyway*, so it could always lie about what key was used by faking
 	// the signature itself.
 	if err = jws.Verify(); err != nil {
-		return nil, zeroKey, err
+		return nil, zeroKey, "", err
 	}
 
-	// TODO Return JWS body
-	return []byte(jws.Payload), jws.Header.Key, nil
-}
-
-// The ID is always the last slash-separated token in the path
-func parseIDFromPath(path string) string {
-	re := regexp.MustCompile("^.*/")
-	return re.ReplaceAllString(path, "")
-}
-
-// Problem objects represent problem documents, which are
-// returned with HTTP error responses
-// https://tools.ietf.org/html/draft-ietf-appsawg-http-problem-00
-type ProblemType string
+	var regID string
+	if regCheck {
+		// Check that the key is assosiated with an actual account
+		reg, err := wfe.SA.GetRegistrationByKey(jws.Header.Key)
+		if err != nil {
+			return nil, zeroKey, "", err
+		}
+		regID = reg.ID
+	}
 
-type problem struct {
-	Type     ProblemType `json:"type,omitempty"`
-	Detail   string      `json:"detail,omitempty"`
+	// TODO Return JWS body
+	return []byte(jws.Payload), jws.Header.Key, regID, nil
```

Comment:
Let's actually return the full registration as long as we've got it, not just the ID.